pkg/labels/labels.go

```diff
@@ -74,6 +74,14 @@ const (
 	// Boolean value which can be parsed with strconv.ParseBool() is required.
 	// (like "nerdctl/bypass4netns=true" or "nerdctl/bypass4netns=false")
 	Bypass4netns = Prefix + "bypass4netns"
+
+	// NetworkNamespace is the network namespace path to be passed to the CNI plugins.
+	// When this annotation is set from the runtime spec.State payload, it takes
+	// precedence over the PID based resolution (/proc/<pid>/ns/net) where pid is
+	// spec.State.Pid.
+	// This is mostly used for VM based runtime, where the spec.State PID does not
+	// necessarily lives in the created container networking namespace.
+	NetworkNamespace = Prefix + "network-namespace"
```

Comment:
IIUC, this doesn't work with `nerdctl run --restart=always` containers when the host OS rebooted, because the specified netns no longer exists

Reply:
I am not familiar with that part of the code base, so please correct me if/where I'm wrong:

After a reboot, nerdctl will create a  new containerd task, which will trigger kata to create a new netns and update the `spec.State.Annotations` accordingly. The OCI hooks will be called with that newly created netns, and nerdctl will use it to run the CNI plugins.

What am I missing?

Reply:
>  nerdctl will create a new containerd task

The new task is created by the restart monitor, not by nerdctl (unless user executes nerdctl manually): https://github.com/containerd/containerd/tree/v1.6.0-rc.4/runtime/restart/monitor

Reply:
If you tested it with the restart monitor and it works, this PR LGTM, but the annotation probably shouldn't be defined in `pkg/labels`, as it seems pure OCI annotation, not containerd label.

Reply:
I started a container with `--restart always` and verified that it was up and running after a reboot, with networking working as well.

> the annotation probably shouldn't be defined in pkg/labels, as it seems pure OCI annotation, not containerd label.

You mean I should use a `org.opencontainers` prefixed annotation?

Reply:
> The proposed annotation is not a containerd label, so it shouldn’t be defined in `pkg/labels ` package, but should be probably defined in `pkg/ocihook` package.
> 
> The `nerdctl/` prefix is fine, and it must NOT be `org.opencontainers/`, but `io.katacontainers/` might be even better?

Thanks for the explanation, that makes sense now.
Since the main and single consumer of this annotation is nerdctl, I'd rather keep the `nerdctl/` prefix. I'll move the definition to the `ocihook` package.

Reply:
I changed the first commit to move the annotation key under `ocihook.go`. Let me know if that's fine with you.